src/errors/BaseError.ts

```diff
@@ -15,32 +15,16 @@ class BaseIsomerError extends Error implements IsomerInternalError {
 
   isV2Err = false // indicates if this is the new error format
 
-  componentCode: string // a selection from ComponentTypes
-
-  fileCode: string // a selection from FileCodes
-
-  constructor({
-    status = 500,
-    code = "BaseError",
-    message = "Something went wrong",
-    meta = {},
-    componentCode = ComponentTypes.Other,
-    fileCode = "000",
-  }) {
+  constructor({ status = 500, message = "Something went wrong", meta = {} }) {
     super()
     Error.captureStackTrace(this, this.constructor)
     this.name = this.constructor.name
-    this.code = code
     this.status = status
+    this.code = this.name
```

Comment:
if `this.code` = `this.name`, why do we have 2 variables here ah? 

[nit]: could we rename `code` to `errorType` instead?

Reply:
tbh, this is just to keep parity with the doc - because implementing engineers will (probably) refer to the doc, i opted to keep the `code` argument. honestly, not a big deal - i'm open to removing it and we refer to it as `name` instead.

re: `code` vs `errorType` - sure, wdyt about just keeping as `name`? i think this avoids the most important issue, which is conflation of properties between teh internal/external rep

Reply:
ye prefer to change doc when needed than keep two variables that does the same thing 

> re: code vs errorType - sure, wdyt about just keeping as name? i think this avoids the most important issue, which is conflation of properties between teh internal/external rep
sure